code/modules/antagonists/heretic/knowledge/sacrifice_knowledge/sacrifice_knowledge.dm

```diff
@@ -368,7 +368,13 @@
 	sac_target.remove_status_effect(/datum/status_effect/unholy_determination)
 	sac_target.reagents?.del_reagent(/datum/reagent/inverse/helgrasp/heretic)
 	sac_target.clear_mood_event("shadow_realm")
-	sac_target.gain_trauma(/datum/brain_trauma/mild/phobia/supernatural, TRAUMA_RESILIENCE_MAGIC)
+	if(!(IS_HERETIC(sac_target)))
+		sac_target.gain_trauma(/datum/brain_trauma/mild/phobia/heresy, TRAUMA_RESILIENCE_MAGIC)
+
+	if(IS_HERETIC(sac_target))
+		REMOVE_TRAIT(sac_target, TRAIT_ALLOW_HERETIC_CASTING, src)
```

Comment:
Removing this trait wouldn't prevent the heretic from using spells. The heretic doesn't have it by default(so in some cases you are removing a trait which the heretic doesn't even have), and it is gained from wearing a focus item. So even if you will remove it, it can be easily gained back by dropping and putting back a focus item. Additionaly, you should do it like REMOVE_TRAIT(something, trait define, the trait source), because if you do it like you do you will try to remove a trait granted by the sacrifice ritual, which doesn't grant this trait.